pkg/controller/mysqldatabase/db_controller.go

```diff
@@ -94,9 +94,13 @@ func (r *ReconcileMySQLDatabase) Reconcile(ctx context.Context, request reconcil
 
 	// Check if the resource is deleted
 	if !db.ObjectMeta.DeletionTimestamp.IsZero() {
-		err = r.deleteDatabase(ctx, db)
-		if err != nil {
-			return reconcile.Result{}, err
+		if db.ObjectMeta.Annotations != nil && mysqlv1alpha1.MysqlResourceDeletionPolicy(db.ObjectMeta.Annotations[mysqlv1alpha1.MysqlResourceDeletionPolicyAnnotationKey]) == mysqlv1alpha1.MysqlResourceDeletionPolicyRetain {
```

Comment:
you can add a helper function to simplify this part

Reply:
yes ,LGTM